File: wlr/SPX.Paths/PathFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OGUR.Collision;

namespace OGUR.Paths
{
    class PathFactory
    {
        private static readonly Path[] s_paths = new Path[50];
        private static int s_pathIndex = 0;

        public static Path Create(Point2 source,Point2 dest)
        {
            return GetNext().Reset(source, dest);
        }

        public static Path Create(Path path)
        {
            return GetNext().Copy(path);
        }

        private static Path GetNext()
        {
            if (s_paths[0] == null)
            {
                for (int ii = 0; ii < s_paths.Length; ii++)
                {
                    s_paths[ii] = new Path();
                }
            }
            s_pathIndex = (++s_pathIndex) % s_paths.Length;
            return s_paths[s_pathIndex];
        }
    }
}

File: proj/src/spx/paths/PathFactory.java
package spx.paths;

import spx.core.Point2;

class PathFactory {
    private static final Path[] __paths = new Path[50];
    private static int __pathIndex = 0;

    public static Path create(Point2 source, Point2 dest) {
        return getNext().reset(source, dest);
    }

    public static Path create(Path path) {
        return getNext().copy(path);
    }

    private static Path getNext() {
        if (__paths[0] == null) {
            for (int ii = 0; ii < __paths.length; ii++) {
                __paths[ii] = new Path();
            }
        }
        __pathIndex = (++__pathIndex) % __paths.length;
        return __paths[__pathIndex];
    }
}
